pkg/workflows/manager.go

```diff
@@ -0,0 +1,46 @@
+/*
+This file is part of REANA.
+Copyright (C) 2022 CERN.
+
+REANA is free software; you can redistribute it and/or modify it
+under the terms of the MIT License; see LICENSE file for more details.
+*/
+
+package workflows
+
+import (
+	"reanahub/reana-client-go/client"
+	"reanahub/reana-client-go/client/operations"
+	"reanahub/reana-client-go/pkg/config"
+	"reanahub/reana-client-go/pkg/validator"
+)
+
+// UpdateStatus updates the status of the specified workflow.
+func UpdateStatus(
+	token, workflow, status string,
+	includeWorkspace, includeAllRuns bool,
+) error {
+	if err := validator.ValidateChoice(status, config.GetRunStatuses(true), "status"); err != nil {
+		return err
+	}
+
+	deleteParams := operations.NewSetWorkflowStatusParams()
+	deleteParams.SetAccessToken(&token)
+	deleteParams.SetWorkflowIDOrName(workflow)
+	deleteParams.SetStatus(status)
+	deleteParams.SetParameters(operations.SetWorkflowStatusBody{
+		AllRuns:   includeAllRuns,
+		Workspace: includeWorkspace,
+	})
+
+	api, err := client.ApiClient()
+	if err != nil {
+		return err
+	}
+	_, err = api.Operations.SetWorkflowStatus(deleteParams)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
```

Comment:
I decided to put this function here because it can be used by other commands. However, the test coverage is decreased, since this is only tested in `TestDelete` in another package